app/scripts/modules/core/src/function/function.write.service.ts

```diff
@@ -0,0 +1,49 @@
+import { IPromise } from 'angular';
+
+import { Application } from 'core/application/application.model';
+import { ITask } from 'core/domain';
+import { IJob, TaskExecutor } from 'core/task/taskExecutor';
+
+export interface IFunctionUpsertCommand extends IJob {
+  functionName: string;
+  cloudProvider: string;
+  credentials: string;
+  description?: string;
+  region: string;
+  operation: string;
+}
+
+export interface IFunctionDeleteCommand extends IJob {
+  cloudProvider: string;
+  functionName: string;
+  credentials: string;
+  region: string;
+  vpcId?: string;
+  operation?: string;
+}
+
+export class FunctionWriter {
+  public static deleteFunction(command: IFunctionDeleteCommand, application: Application): ng.IPromise<ITask> {
+    command.type = 'lambdaFunction';
+    command.operation = 'deleteLambdaFunction';
+    return TaskExecutor.executeTask({
+      job: [command],
+      application,
+      description: `Delete Function: ${command.functionName}`,
+    });
+  }
+
+  public static upsertFunction(
+    command: IFunctionUpsertCommand,
+    application: Application,
+    descriptor: string,
+    params: any = {},
+  ): IPromise<ITask> {
+    Object.assign(command, params);
+    return TaskExecutor.executeTask({
+      job: [command],
+      application,
+      description: `${descriptor} Function: ${command['functionName']}`,
```

Comment:
Can we use dot notation and add a fallback name so it isn't interpolated to "undefined"?
`${command.functionName || ''}`

Reply:
done. Will be seen in the next push.